pages/_app.tsx

```diff
@@ -19,6 +23,12 @@ export interface MyAppProps extends AppProps {
 export default function MyApp(props: MyAppProps) {
   const { Component, emotionCache = clientSideEmotionCache, pageProps } = props;
 
+  const prefersDarkMode = useMediaQuery('(prefers-color-scheme: dark)');
```

Comment:
Let's move this logic + `React.useMemo` into a separate file under a specific directory (e.g. hooks). Last but not least, please, use restructuring:
`import React from 'react';` -> `import { useMemo } from 'react';`

Reply:
'+